src/gsy_e/models/leaves.py

```diff
@@ -74,20 +96,32 @@ def __init__(self, name, config, uuid=None, **kwargs):
                 except KeyError:
                     logging.error(f"{self.strategy_type} could not be found "
                                   f"in external_strategies_mapping, using template strategy.")
+
+        # Gather all constructor arguments from all base classes of the strategy class.
+        allowed_arguments = ['self']
+        for strategy_base in self.strategy_type.__mro__:
+            if '__init__' in strategy_base.__dict__:
+                allowed_arguments += inspect.getfullargspec(strategy_base).args[1:]
+
+        # Filter out the arguments that are not accepted by any constructor in the strategy
+        # class hierarchy and log them.
+        not_accepted_args = set(kwargs.keys()).difference(allowed_arguments)
+        if not_accepted_args:
+            logging.warning(f"Trying to construct area strategy {name} with not allowed "
+                            f"arguments {not_accepted_args}")
```

Comment:
Could you please describe why this is needed? It seems a bit "hacky" so I'd find it good to have some documentation that described why this is needed.

(Nit on a second note: these lines are not using `"` and lazy formatting)

Reply:
The reason for its existence is the deprecation of the Strategy.parameters class member. In the previous implementation, all the accepted parameters of a strategy were defined by the `parameters` tuple / list that included the input parameters in string format. I personally find this very bad practice, because the constructor arguments need to also be duplicated in the parameters tuple in order to be taken into account to the serializer / deserializer. Therefore I opted to remove the `parameters` tuple from all strategies and try to read the constructor arguments in order to generate the constructor argument list.

Unfortunately, inheritance makes this a more daunting task than expected. To provide the exact example where this fails, is for the ExternalStrategies. These do not define constructors and they rely on the default constructor, which uses as arguments *args and **kwargs and relays them to the base class constructors. Therefore this makes it impossible to find out the constructor arguments just by inspecting this class. The solution that I relied on was MRO: essentially traversing the class hierarchy in order to gather all the constructor arguments from all the subclasses. This works, and the constructor argument list is created, from which we can exclude the arguments that cannot be used for the selected strategy.

I admit it is much less straightforward than explicitly defining the parameters tuple, the advantage of it is that we do not have to maintain these tuples and keep them synced with the constructors. 

WRT the second note, I think I forgot to run pylint in this module and it is a good opportunity to do so now, thanks!

Reply:
> Update: we're actually using this code just once, so it's not strictly necessary. Also, if the SCM strategies don't share the same base class with the "normal" strategies we would need to duplicate this code, and it's probably not worth it.

Thanks for saving me a comment to denote exactly the latter, that the SCM strategies have a different base class and the method needs to be duplicated. 

Also adding a long comment to explain the rationale behind this, thanks!